src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -28,6 +30,11 @@ internal class AnalysisScope
         /// </summary>
         public IEnumerable<SyntaxTree> SyntaxTrees { get; }
 
+        /// <summary>
+        /// Non-source files on which we need to perform analysis.
+        /// </summary>
+        public IEnumerable<AdditionalText> NonSourceFiles { get; }
```

Comment:
Why did u choose the terminology `NonSourceFiles` here vs. say `AdditionalFiles`?

Reply:
I was first thinking of adding support for both AdditionalFile actions and AnalyzerConfigFile actions, but latter needs more work/public API, so reverted back to just additional files in this PR. I can change this to AdditionalFiles, and when we get to AnalyzerConfigFiles, we can rename it as appropriate.